amida-tech/cql-execution
# Depression Screening and Follow-Up for Adolescents and Adults (CQL)

# Related Tickets



- [SAR-75](https://jira.amida-tech.com/browse/SAR-94)

# Other Repos' PR(s) Intended to Work With This PR

# How Things Worked (or Didn't) Before This PR
There was no CQL and associated files for Depression Screening and Follow-Up for Adolescents and Adults

# How Things Work Now (And How to Test)
## How it works now
I've added the CQL and associated files for Depression Screening and Follow-Up for Adolescents and Adults

## To test
run `yarn` to install the necessary packages
run `node exec-files/exec-depression-screening.js `
Use your favorite diffchecker to confirm your output for patientResults and unfilteredResults matches what is below - you can ignore everything else.
```javascript
{ 'numer-EXAMPLE1-FHIR4':
   { 'First Predecessor Year':
      Interval { low: [Date], high: [Date], lowClosed: true, highClosed: false },
     'Lookback Interval One More Year':
      Interval { low: [Date], high: [Date], lowClosed: true, highClosed: false },
     Patient:
      FHIRObject {
        identifier: [Getter],
        active: [Getter],
        name: [Getter],
        telecom: [Getter],
        gender: [Getter],
        birthDate: [Getter],
        deceased: [Getter],
        address: [Getter],
        maritalStatus: [Getter],
        multipleBirth: [Getter],
        photo: [Getter],
        contact: [Getter],
        communication: [Getter],
        generalPractitioner: [Getter],
        managingOrganization: [Getter],
        link: [Getter],
        text: [Getter],
        contained: [Getter],
        extension: [Getter],
        modifierExtension: [Getter],
        id: [Getter],
        meta: [Getter],
        implicitRules: [Getter],
        language: [Getter] },
     'Initial Population': true,
     'Bipolar Disorder Reports': [ [FHIRObject], [FHIRObject] ],
     'Denominator Exclusion': true,
     'Denominator 1': 1,
     'Numerator 1': 1,
     'Positive Depression Screening During Follow-Up Period': [ [FHIRObject] ],
     'Denominator 2': 1,
     'First Positive Depression Screening During Follow-Up Period': [ [FHIRObject] ],
     'Dispensed Antidepressant Within 31 Days': [ [FHIRObject] ],
     'Follow-Up Care Encounters': [ [FHIRObject], [FHIRObject], [FHIRObject] ],
     'Follow-Up Encounter Within 31 Days': [],
     'Same Day Behavioral Encounter': [ [FHIRObject] ],
     'Negative Depression Screening Within 31 Days': [ [FHIRObject] ],
     'Numerator 2': 1,
     'Stratifier 1': false,
     'Stratifier 2': true,
     'Stratifier 3': false,
     'Stratifier 4': false },
  'numer-EXAMPLE2-FHIR4':
   { 'First Predecessor Year':
      Interval { low: [Date], high: [Date], lowClosed: true, highClosed: false },
     'Lookback Interval One More Year':
      Interval { low: [Date], high: [Date], lowClosed: true, highClosed: false },
     Patient:
      FHIRObject {
        identifier: [Getter],
        active: [Getter],
        name: [Getter],
        telecom: [Getter],
        gender: [Getter],
        birthDate: [Getter],
        deceased: [Getter],
        address: [Getter],
        maritalStatus: [Getter],
        multipleBirth: [Getter],
        photo: [Getter],
        contact: [Getter],
        communication: [Getter],
        generalPractitioner: [Getter],
        managingOrganization: [Getter],
        link: [Getter],
        text: [Getter],
        contained: [Getter],
        extension: [Getter],
        modifierExtension: [Getter],
        id: [Getter],
        meta: [Getter],
        implicitRules: [Getter],
        language: [Getter] },
     'Initial Population': false,
     'Bipolar Disorder Reports': [ [FHIRObject], [FHIRObject] ],
     'Denominator Exclusion': true,
     'Denominator 1': 0,
     'Numerator 1': 0,
     'Positive Depression Screening During Follow-Up Period': [ [FHIRObject] ],
     'Denominator 2': 0,
     'First Positive Depression Screening During Follow-Up Period': [ [FHIRObject] ],
     'Dispensed Antidepressant Within 31 Days': [ [FHIRObject] ],
     'Follow-Up Care Encounters': [ [FHIRObject], [FHIRObject], [FHIRObject] ],
     'Follow-Up Encounter Within 31 Days': [],
     'Same Day Behavioral Encounter': [ [FHIRObject] ],
     'Negative Depression Screening Within 31 Days': [ [FHIRObject] ],
     'Numerator 2': 0,
     'Stratifier 1': false,
     'Stratifier 2': false,
     'Stratifier 3': false,
     'Stratifier 4': false } }
{ 'Denominator Count': 2,
  'Numerator Count': 2,
  'CDC Measure Score': 1 }
```
## To Review
Here's what to look for in each different file:

`exec-depression-screening.js` - This is the exec file that actually does the work - there's not much to review here.

`depression-screening-follow-up-codes.js` - The list of codes in each value set. This is programmatically generated using the CQL from the [VSAC](https://vsac.nlm.nih.gov/) API, so there's nothing to review here

`depression-screening.json` - JSON version of the CQL created using the [clinical_quality_language](https://github.com/cqframework/clinical_quality_language) - nothing to review here

`depression-screening-follow-up.json` - This contains the fake patients and their data. If you are really gung-ho, you can review the output of the results and confirm that the expressions with the CQL library returned the correct response based on the data, but that might take a while.

`depression-screening.cql` - this is the main piece of the PR. Read through the following requirements, and as best you can, confirm that the CQL matches the guidelines. If you'd prefer, whenever you get to this portion I can send you the word doc these requirements are in, which might be easier than looking at screenshots.

### Use `K. Health Plan Description AND ECDS_pp 557-650.docx` in the HEDIS textbook `pages 602-606`
